website/upgrade-guides/0-12.html.markdown

```diff
@@ -39,80 +92,341 @@ Examples of these workarounds include:
+ambiguous, in which case the tool will add to your configuration a comment
+containing the marker `TF-UPGRADE-TODO` to indicate a situation where your
+human intuition is required to decide how to proceed.
 
-Terraform 0.12 will be released with a migration tool that will make most of
-the required updates automatically, and also provide guidance on any changes
-that require human input.
+We recommend running the upgrade tool in a clean version control work tree so
+that you can use the VCS diffing tools to easily see and review all of the
+proposed updates. Search the upgraded module for `TF-UPGRADE-TODO` to find
+the situations where human attention is required.
 
-For users who follow the examples in the Terraform documentation, there should
-be no required changes. However, we still recommend to run the migration tool
-to upgrade to the more readable syntax conventions supported in this release,
-and to draw attention to any potential issues.
+Even if your existing configuration works without upgrading, we still recommend
+to run the upgrade tool to update to the more readable syntax conventions
+supported in this release, and to draw attention to any potential issues.
 
-## Remote State Referencing
+The following sections describe in more detail some of the situations that will
+be detected and upgraded by the upgrade tool, both to help understand the
+purpose of certain proposed changes and to help users who may not wish to
+use the automatic upgrade tool. However, the following sections are not
+completely comprehensive so we still recommend using the upgrade tool to review
+its output, even if you then discard the proposed changes and make your updates
+manually.
 
-`terraform` provider has changed schema which caused all outputs to be nested
-under `outputs` field.
+### Remote state references
 
-### Before
+The `terraform_remote_state` data source has changed slightly for the v0.12
+release to make all of the remote state outputs available as a single map
+value, rather than as top-level attributes as in previous releases.
+
+In previous releases, a reference to a `vpc_id` output exported by the remote
+state data source might have looked like this:
 
 ```hcl
-data.terraform_remote_state.vpc.something
+data.terraform_remote_state.vpc.vpc_id
 ```
 
-### After
+This value must now be accessed via the new `outputs` attribute:
 
 ```hcl
-data.terraform_remote_state.vpc.outputs.something
+data.terraform_remote_state.vpc.outputs.vpc_id
 ```
 
-## Upgrading Terraform providers
+Where appropriate, you can also access the outputs attribute directly to
+work with the whole map as a single value:
+
+```hcl
+data.terraform_remote_state.vpc.outputs
+```
+
+### Attributes vs. blocks
+
+Terraform resource configurations consist of both arguments that set
+individual properties of the main object being described, and nested blocks
+which declare zero or more other objects that are modelled as being part of
```

Comment:
🤔  modelled is the british spelling, modeled is american english - I don't know if we have a stated guideline on this but in general it seems like we favor american english in our documentation (I don't have any feelings or opinions on this).

Reply:
Indeed, it should be American English. My fault for being British!